src/Scope.php

```diff
@@ -25,11 +25,15 @@ class Scope
 
     /**
      * @param Scope $scope
+     * @param string $key - an optional key. defaults to the scope's object hash
      */
-    public function peridotAddChildScope(Scope $scope)
+    public function peridotAddChildScope(Scope $scope, $key = "")
     {
         $scope->peridotSetParentScope($this);
-        $this->peridotGetChildScopes()->attach($scope);
+        if (empty($key)) {
+            $key = spl_object_hash($scope);
+        }
+        $this->peridotChildScopes[$key] = $scope;
```

Comment:
Is there any use case where we would want to return the generated hash key if one was not supplied?  Maybe not, since if you cared about the key, you should have provided it yourself?